rfc/text/basic/bp_04_sessions.md

```diff
@@ -333,6 +333,145 @@ and the other peer replies
 
 **Difference between ABORT and GOODBYE**
 
-The differences between `ABORT` and `GOODBYE` messages is that `ABORT` is never replied to by a Peer, whereas `GOODBYE` must be replied to by the receiving Peer.
+The differences between `ABORT` and `GOODBYE` messages is that `ABORT` is never replied to by a Peer, whereas `GOODBYE` must be replied to by the receiving Peer.
 
 > Though `ABORT` and `GOODBYE` are structurally identical, using different message types serves to reduce overloaded meaning of messages and simplify message handling code.
+
+
+## All Together
+
+### Correct Exapmle
+
+Here's a complete example showing how a client establishes and closes a session.
+
+(Client to Router) TLS Handshake Request
+
+{align="left"}
+    CONNECT example.net:443 HTTP/1.1
+    Host: example.net:443
+
+(Router to Client) TLS Handshake Response
+
+{align="left"}
+    HTTP/1.1 200 Connection established
+
+(Client to Router) WebSocket Handshake
+
+{align="left"}
+    GET / HTTP/1.1
+    Upgrade: websocket
+    Host: example.net
+    Origin: https://example.net
+    Sec-WebSocket-Key: ObmKzXaAVXFLmF7G6AlOZA==
+    Sec-WebSocket-Version: 13
+    Connection: Upgrade
+    Sec-WebSocket-Protocol: wamp.2.json
+
+(Router to Client) WebSocket Accept
+
+{align="left"}
+    HTTP/1.1 101 Switching Protocols
+    Date: Fri, 23 Mar 2012 16:45:00 GMT
+    Connection: upgrade
+    Upgrade: websocket
+    Sec-WebSocket-Accept: ftraiO6TrSF1uX+qeMp4tQ74zts=
+    Sec-WebSocket-Protocol: wamp.2.json
+    server: DemoRouter
+
+(Client to Router) Client Hello
+
+{align="left"}
+    [1, "org.example", {"agent": "WAMP/2.0 (Client)", "roles": {"publisher": {}, "subscriber": {}}}]
+
+(Router to Client) Router Welcome
+
+{align="left"}
+    [2, 5606911429535183, {"agent": "WAMP/2.0 (Router)", "roles": {"broker": {}}}]
```

Comment:
Regarding schema I mean something like that:

```suggestion
    // [WELCOME, Session|id, Details|dict]
    [2, 5606911429535183, {"agent": "WAMP/2.0 (Router)", "roles": {"broker": {}}}]
```

Reply:
Right, I'll do it, but I've no chance to do it.